src/gui/inputdevices/PenInputHandler.cpp

```diff
@@ -142,13 +142,33 @@ auto PenInputHandler::actionStart(InputEvent const& event) -> bool {
     return false;
 }
 
+double PenInputHandler::inferPressure(PositionInputData const& pos, XojPageView* page) {
```

Comment:
Two things with infering pressure. I generally like the idea behind it, as drawing quickly is often associated with less pressure. But not all users might like this as it could be understood as unintuitive. Hence I'd appreciate it if you could make this optional via a preference in the preferences dialog.

Additionally this feature is quite specific to the TouchDrawingInputHandler and should therefore reside in this class. If you add it to the PenInputHandler, you would block any new tool that might actually not want pressure by design.

Reply:
I think it might make sense for pressure to be enabled for multiple input sources. For example, mice also lack pressure sensitivity.

I have added this to the preferences dialog.